instrumentation/opentelemetry-instrumentation-django/src/opentelemetry/instrumentation/django/middleware/otel_middleware.py

```diff
@@ -447,12 +447,19 @@ def process_response(self, request, response):
 def _parse_duration_attrs(
     req_attrs, sem_conv_opt_in_mode=_HTTPStabilityMode.DEFAULT
 ):
-    return _filter_semconv_duration_attrs(
+    # http.target to be included in old semantic conventions
+    target = None
+    if sem_conv_opt_in_mode is _HTTPStabilityMode.DEFAULT:
+        target = req_attrs.get(SpanAttributes.HTTP_TARGET)
```

Comment:
```suggestion
        target = req_attrs.get(SpanAttributes.HTTP_ROUTE)
```

`target` in metrics ~ `route` in spans

Reply:
Not entirely sure what you mean by this? We parsing duration attributes so this pertains to metrics.

Reply:
I think `req_attrs` had a bunch of span attributes as well. Looking more closely I think this was probably fine, and the change you just made feels better still. I'd be more confident that this won't break in future if (1) the tests were stricter and (2) span attributes already had `http.target` as they should.

Reply:
I've added stricter tests that assert values and keys of the attributes instead of just the presence of keys that follow a static list. In regards to (2), I believe span attributes are populated by [wsgi](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/instrumentation/opentelemetry-instrumentation-django/src/opentelemetry/instrumentation/django/middleware/otel_middleware.py#L210) middleware and `http.target` is already [populated](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/instrumentation/opentelemetry-instrumentation-wsgi/src/opentelemetry/instrumentation/wsgi/__init__.py#L348).

Reply:
Thanks, I do think that's an improvement. The span attributes stuff does push the scope here, but if you're willing, I think this would be even better:

```python
expected_attributes = {
    'http.request.method': 'GET',
    'http.method': 'GET',
    'http.url': 'http://testserver/traced/',
    'http.scheme': 'http',
    'url.scheme': 'http',
    'http.status_code': 200,
    'http.response.status_code': 200,
    'server.port': 80,
    'client.address': '127.0.0.1',
    'network.protocol.version': '1.1',
}

if DJANGO_2_2:
    expected_attributes['http.route'] = '^traced/'

self.assertEqual(span.attributes, expected_attributes)
```

That's an example for `test_traced_get_both_semconv`. Then it would be clear whether or not `http.target` is in the span attributes. Right now if it is then the tests don't show it.

Reply:
@alexmojaki 

This is a great point and we most likely need this scenario tested out.

I do not believe `http.target` was ever explicitly supported for `django` and only relies on the `wsgi` and `asgi` implementations of extracting this information. If `http.target` is missing, it simply would be a feature request of the django instrumentation and outside the scope of this pr. We also do not want to be including old semantic convention changes so the feature request would more accurately be "support url.path/url.query" in this case.